### ISSUE TITLE
Changing SVG icon color and size via CSS?

### ISSUE THREAD

**Issue by 4Digits** [filamentgroup/grunticon#235]
Is it possible to change an icon's color and size via CSS?

Currently, the svg icons I generated have an inline width / height declaration, something like:

`<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="32" height="32" viewBox="0 0 32 32">`

as well as a color specification, e.g. `fill="000"`.

I tried to remove color and width/height information from the svg-files and then tried to apply different CSS styles, e.g.:

`.svg-icon {
	display: inline-block;
	width: 68px;
	height: 68px;
	fill: #ff0000;
}` 

However, this is not working. 

Any idea how to achieve different sizes and colors?

**Comment by robtarr** [comment]
You need to put the CSS rule on the `path` element. Try:

```
.svg-icon path {
  fill: #ff0000;
}
```

**Comment by jefflembeck** [comment]
@robtarr thanks!

@4Digits - yeah, the icon needs to be styled on the svg itself in the case of an embedded svg instead of the wrapping div

**Comment by scottjehl** [comment]
So, this depends a bit on the svg markup too, fwiw.

CSS `fill` will cascade from HTML elements into child SVG shapes that don't have a `fill` specified via their svg `fill` attribute. 

### Here's a red CSS fill on `a` element parent:
![image](https://cloud.githubusercontent.com/assets/214783/6444675/c1cf0a8a-c0c4-11e4-8762-2f0cabeef500.png)

### Here's a blue CSS fill on `a` element parent:
![image](https://cloud.githubusercontent.com/assets/214783/6444688/df614e96-c0c4-11e4-8011-31a8c46a89aa.png)

So basically, if you want to style a path fill with css, I think you just need to skip the inline `fill` attribute.

That said, for multi-color svgs, you can add classes to particular shapes in the SVG to style them as needed. The icons on our demo page do that a lot. (See "can I apply CSS to them?" http://www.grunticon.com )

Lastly, if you want the color of an element's text to cascade, you can use the `currentColor` keyword to varying effect (support varies I think). More here: https://css-tricks.com/cascading-svg-fill-color/

**Comment by oleole90** [comment]
TL;DR: Also make sure there are **no** `fill="xxxx"` attribute in your svg markup.

**Comment by dominikwilkowski** [comment]
You can also use `fill="currentcolor"` and then color the icon via the `color` attribute in the parent much like we used to do it with icon fonts...

**Comment by animanoir** [comment]
@dominikwilkowski That worked for me.

**Comment by tibcom** [comment]
You can use `zoom` parameter. For example `zoom:0.5` for 50% of original size of your icon.
